ambry-rest/src/main/java/com/github/ambry/rest/RestServer.java

```diff
@@ -176,13 +176,14 @@ public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterM
    * @param sslFactory the {@link SSLFactory} to be used. This can be {@code null} if no components require SSL support.
    * @param addedChannelHandlers a list of {@link ChannelHandler} to add to the {@link io.netty.channel.ChannelInitializer} before
    *                             the final handler.
+   * @param addedQuotaEnforcers a list of {@link QuotaEnforcer}s to add to the {@link QuotaManagerFactory}.
    * @param reporterFactory if non-null, use this function to set up a {@link JmxReporter} with custom settings. If this
    *                        option is null the default settings for the reporter will be used.
    * @throws InstantiationException if there is any error instantiating an instance of RestServer.
    */
   public RestServer(VerifiableProperties verifiableProperties, ClusterMap clusterMap,
       NotificationSystem notificationSystem, SSLFactory sslFactory, List<ChannelHandler> addedChannelHandlers,
-      Function<MetricRegistry, JmxReporter> reporterFactory) throws Exception {
+      List<QuotaEnforcer> addedQuotaEnforcers, Function<MetricRegistry, JmxReporter> reporterFactory) throws Exception {
```

Comment:
So `addedQuotaEnforcers` is added here to integrate with internal code?

Reply:
Yes.. this is for cases where quota enforcers need to be specified outside quotaManager. Earlier we were just putting empty list instead which defeated the purpose.

Note that for now we have figured out a way to use the closed source code without needing this. But this is a bug anyways because if we always assign it to empty list, then this field is basically not needed.
For now we are not using it anywhere. But I want to keep it for easy extension in open source code.

Reply:
why do we still have additional enforcer from outside of the configuration?

Reply:
I want to keep it for easy extension in open source code. I will create a followup ticket to remove it in future if we see no  value in it. Its not being used right now.